pkg/objstore/s3/s3.go

```diff
@@ -67,8 +67,8 @@ var DefaultConfig = Config{
 		MaxConnsPerHost:       0,
 	},
 	// Minimum file size after which an HTTP multipart request should be used to upload objects to storage.
-	// Set to 128 MiB as in the minio client.
-	PartSize: 1024 * 1024 * 128,
+	// Set to 16 MiB as in the minio client.
+	PartSize: 1024 * 1024 * 16,
```

Comment:
Let's be careful here. We don't want such small parts to be uploaded as multipart. We want 128MB multi-parts. But it does not mean that it should hold 128MB or more in memory, so I would fix minio client rather 🤔